pkg/controller/manager/sync_test.go

```diff
@@ -4,675 +4,829 @@ package manager
+								StartDate: pointer.Ptr(validDate),
+							},
 							Name: "backup",
-							SchedulerTaskSpec: scyllav1.SchedulerTaskSpec{
-								StartDate: pointer.Ptr("2021-01-01T11:11:11Z"),
+							ID:   pointer.Ptr("backup-id"),
+							Labels: map[string]string{
+								"scylla-operator.scylladb.com/managed-hash": "0aeio9mEaSKenZ7/GRW4l0TFm7f9kY2w7A3wpO3du5+EfjM9zIqJunon9vT+VvGNcwYxQQqF4gmZW1GSLpZU6g==",
 							},
 						},
-						Location:  []string{"s3:backup"},
-						Retention: 3,
+						Location: []string{"gcs:location"},
 					},
 				},
-			},
-			Status: scyllav1.ScyllaClusterStatus{
-				ManagerID: pointer.Ptr(clusterID),
-			},
-			State: state{
-				Clusters: []*managerclient.Cluster{{
-					ID:        clusterID,
-					Name:      clusterName,
-					AuthToken: clusterAuthToken,
-				}},
 				RepairTasks: map[string]scyllav1.RepairTaskStatus{
 					"repair": {
 						TaskStatus: scyllav1.TaskStatus{
 							SchedulerTaskStatus: scyllav1.SchedulerTaskStatus{
-								StartDate: pointer.Ptr("2006-01-02T15:04:05Z"),
+								StartDate: pointer.Ptr(validDate),
 							},
 							Name: "repair",
 							ID:   pointer.Ptr("repair-id"),
 							Labels: map[string]string{
-								"scylla-operator.scylladb.com/managed-hash": "JcrUPldfq9/FT/tAaXzdY2aclZFsjTlRsYDh7LEjM3K5XRbl8w+jUGvZdBHRRSZ28TdWu2dsa/L5LBxxWjIpHw==",
+								"scylla-operator.scylladb.com/managed-hash": "Il1bCvnHQAs7fjjuE4OHn6Au6tvMxNsVrfvdbwg3uYeNZPzYR+qgvDyAEjVrwakzfOTu/e+viMTWzaCRLqYetA==",
 							},
 						},
+						Intensity:           pointer.Ptr("0.5"),
+						Parallel:            pointer.Ptr[int64](0),
 						SmallTableThreshold: pointer.Ptr("1GiB"),
-						Intensity:           pointer.Ptr("1"),
-						Parallel:            pointer.Ptr[int64](1),
 					},
 				},
-				BackupTasks: map[string]scyllav1.BackupTaskStatus{
-					"backup": {
-						TaskStatus: scyllav1.TaskStatus{
-							SchedulerTaskStatus: scyllav1.SchedulerTaskStatus{
-								StartDate: pointer.Ptr("2006-01-02T15:04:05Z"),
-							},
-							Name: "backup",
-							ID:   pointer.Ptr("backup-id"),
-							Labels: map[string]string{
-								"scylla-operator.scylladb.com/managed-hash": "NpboZYiDmzZfS84omU0kgZxxDzg5p3IEhKCWU0sS6G0QpSh2KZFPHB7AlJohyqo+RjBG2aEIqbBW8GiDo18uxw==",
-							},
-						},
-						Location:  []string{"s3:other-backup"},
-						Retention: pointer.Ptr[int64](1),
-					},
+			},
+			expectedActions: []action{
+				&deleteTaskAction{
+					ClusterID: "bead6247-d9e4-401c-84b4-ad0bffe36eac",
+					TaskType:  "repair",
+					TaskID:    "repair-id",
+				},
+				&deleteTaskAction{
+					ClusterID: "bead6247-d9e4-401c-84b4-ad0bffe36eac",
+					TaskType:  "backup",
+					TaskID:    "backup-id",
 				},
 			},
-			Actions: nil,
+			expectedRequeue: false,
+			expectedErr:     nil,
 		},
 	}
 
-	for _, test := range tcs {
-		t.Run(test.Name, func(t *testing.T) {
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
 			ctx := context.Background()
```

Comment:
nit: add sane timeout

Reply:
nothing in this function consumes the context, it's just legacy, so I'd be leaning towards removing the param if anyhing - wdyt?

Reply:
I would be against removing context parameter, it's useful once its has its user and having it is recommended practice. I'm fine not touching the timeout too.